### ISSUE TITLE
Ammo in classic mode should stack

### ISSUE THREAD

**Issue by lambda-11235** [red-eclipse/world#104]
Currently in classic mode when you pick up a weapon you are holding you discard it, effectively switching ammo capacities. This is kind of annoying with the more limited ammo we have now, as you have to guess if the weapon your picking up has more ammo. Plus, it's counter intuitive to the other modes. I propose that ammo should stack like in other modes.

**Comment by qreeves** [comment]
Define stack?

**Comment by molexted** [comment]
E.g. if I have a weapon with 20 bullets in inventory and pick up a classic weapon of the same type with e.g. 30 ammo included, I would expect the bullets to stack (have 50 bullets) rather than keep the new gun and throw out the old gun with its ammo (leaving me with 30 bullets and discarding 20 onto the ground).

**Comment by Dziq84** [comment]
yeah its a kind of strange behaviour now that ammo get discarded along with the gun, just to replace it with the same gun type
